### PR TITLE
Feature/s3integration

### PR DESCRIPTION
**Pull request by sinaure**
s3 integration via secret creds

### CODE REVIEW

**Comment by deadeyegoodwin** [comment]
@sinaure  This seems to replace your earlier PR, is that correct?
Please check https://github.com/triton-inference-server/server/blob/master/CONTRIBUTING.md#contributor-license-agreement-cla for the CLA and submit it as instructed so that we can accept this submission.

**Review by deadeyegoodwin — CHANGES_REQUESTED** [review]
Change is generally OK but don't change the default behavior of the existing as it is referenced in the README. You should add a section to the README describing how the S3 configuration can be used.

File: deploy/single_server/templates/s3-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{{- if .Values.s3.enabled }}
```
[the remark this answers is not in the document]
Comment (sinaure) [reply deploy/single_server/templates/s3-secret.yaml:1]: line 62

**Comment by rmccorm4** [comment]
Closing as we have an AWS/S3 example that supports secrets: https://github.com/triton-inference-server/server/tree/main/deploy/aws.

**Pull request closed without merging**